src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -322,20 +321,16 @@ private BaseResponseBean doSyncGetNext(FormplayerSyncScreen nextScreen,
     }
 
     private NotificationMessage doSync(FormplayerSyncScreen screen) throws Exception {
-        ResponseEntity<String> responseEntity = syncRequester.makeSyncRequest(screen.getUrl(),
-                screen.getBuiltQuery(),
-                restoreFactory.getUserHeaders());
-        if (responseEntity == null) {
-            return new NotificationMessage("Session error, expected sync block but didn't get one.", true, NotificationMessage.Tag.sync);
-        }
-        if (responseEntity.getStatusCode().is2xxSuccessful()) {
-            // Don't purge for case claim
-            restoreFactory.performTimedSync(false, false);
-            return new NotificationMessage("Case claim successful.", false, NotificationMessage.Tag.sync);
-        } else {
+        try {
+            webClient.post(screen.getUrl(), screen.getQueryParams());
+        } catch (RestClientResponseException e) {
             return new NotificationMessage(
-                    String.format("Case claim failed. Message: %s", responseEntity.getBody()), true, NotificationMessage.Tag.sync);
+                    String.format("Case claim failed. Message: %s", e.getResponseBodyAsString()), true, NotificationMessage.Tag.sync);
+        } catch (RestClientException e) {
+            return new NotificationMessage("Unknown error performing case claim", true, NotificationMessage.Tag.sync);
         }
+        restoreFactory.performTimedSync(false, false);
+        return new NotificationMessage("Case claim successful.", false, NotificationMessage.Tag.sync);
```

Comment:
I don't think this code really worked before since the RestTemplate would raise exceptions if the request was not successful